src/lir/instructions.rs

```diff
@@ -0,0 +1,199 @@
+        expr: Expression,
+    },
+    Store {
+        loc: Loc,
+        dest: Operand,
+        data: Operand,
+    },
+    PushMemory {
+        loc: Loc,
+        res: usize,
+        array: usize,
+        value: Operand,
+    },
+    PopMemory {
+        loc: Loc,
+        res: usize,
+        array: usize,
+    },
+    Constructor {
+        loc: Loc,
+        success: Option<usize>,
+        res: usize,
+        contract_no: usize,
+        constructor_no: Option<usize>,
+        encoded_args: Operand,
+        value: Option<Operand>,
+        gas: Operand,
+        salt: Option<Operand>,
+        address: Option<Operand>,
+        seeds: Option<Operand>,
+        accounts: ExternalCallAccounts<Operand>,
+    },
+
+    LoadStorage {
+        loc: Loc,
+        res: usize,
+        storage: Operand,
+    },
+    ClearStorage {
+        loc: Loc,
+        storage: Operand,
+    },
+    SetStorage {
+        loc: Loc,
+        value: Operand,
+        storage: Operand,
+    },
+    SetStorageBytes {
+        loc: Loc,
+        value: Operand,
+        storage: Operand,
+        offset: Operand,
+    },
+    PushStorage {
+        loc: Loc,
+        res: usize,
+        value: Option<Operand>,
+        storage: Operand,
+    },
+    PopStorage {
+        loc: Loc,
+        res: Option<usize>,
+        storage: Operand,
+    },
+
+    // Call internal function, either static dispatch or dynamic dispatch
+    Call {
+        loc: Loc,
+        res: Vec<usize>,
+        call: InternalCallTy,
+        args: Vec<Operand>,
+    },
+    // Print to log message
+    Print {
+        loc: Loc,
+        operand: Operand,
+    },
+    MemCopy {
+        loc: Loc,
+        src: Operand,
+        dest: Operand,
+        bytes: Operand,
+    },
+
+    ExternalCall {
+        loc: Loc,
+        // Polkadot specific
+        success: Option<usize>,
+        address: Option<Operand>,
+        accounts: ExternalCallAccounts<Operand>,
+        // Solana specific
+        // for deriving and proving the ownership of an account
+        seeds: Option<Operand>,
+        payload: Operand,
+        // Polkadot specific
+        // holding tokens
+        value: Operand,
+        // Polkadot specific
+        // On Solana, charged by transaction
+        gas: Operand,
+        // delegate/regular/static
+        // CallTy is polkadot specific.
+        // It involves difference code generation in emit.
+        callty: CallTy,
+        // only used for analysis passes
+        contract_function_no: Option<(usize, usize)>,
+        // Polkadot specific
+        flags: Option<Operand>,
+    },
+    /// Value transfer; either address.send() or address.transfer()
+    // transfer tokens from one addr to another
+    ValueTransfer {
+        loc: Loc,
+        success: Option<usize>,
+        address: Operand,
+        value: Operand,
+    },
+    /// Self destruct
+    // for destructing the contract from inside
+    // Note: only available on Polkadot
```

Comment:
There seem to be many mixed up comments all over the place, can you please bring them in line (usually doc comments are preferred) :slightly_smiling_face: